### PR TITLE
Account Lockout Feature

### PR DESCRIPTION
**Pull request by starsvoyage**
Added feature that locks user account after failling five log in attempts in a row. Only the Pharmacy Manager (and the superuser) can unlock the account so that the user regains access.

### CODE REVIEW

File: accounts/views.py [hunk 1]
```diff
@@ -20,23 +28,130 @@
 #     return render(request, "register.html", {"form": form})
 
 
+
 def login_view(request):
     if request.method == "POST":
         form = LoginForm(request, data=request.POST)
+        print("Form Data:", request.POST)  # Debugging line
```
Comment (Rad354) [inline accounts/views.py]: These debugging statements can be removed

File: accounts/views.py [hunk 2]
```diff
@@ -20,23 +28,130 @@
...
+                else:
+                    messages.error(request, "Invalid username or password")
+                user_profile.save()
+
     else:
         form = LoginForm()
     return render(request, "login.html", {"form": form})
 
 
+# def login_view(request):
```
Comment (Rad354) [inline accounts/views.py]: This commented function can be deleted if it is no longer used

File: accounts/forms.py [hunk 3]
```diff
@@ -15,4 +16,34 @@
...
+            if user_profile.is_locked:
+                raise forms.ValidationError("This account is locked. Please contact an admin.")
+            
+            # Check if this is the first login
+            if not user_profile.password_changed:
+                self.first_login = True
+
+                    
+            # except User.DoesNotExist:
+            #     raise forms.ValidationError("This username does not exist.")
```
Comment (Rad354) [inline accounts/forms.py]: Do we want this?

**Review by starsvoyage — COMMENTED** [review]
All looks and works great! There are some minor details that we could work out for convenience if we ever had the time to address them.

**Pull request merged**